src/compiler/transformers/declarations/emitResolver.ts

```diff
@@ -122,13 +118,39 @@ export function createEmitDeclarationResolver(file: SourceFile): IsolatedEmitRes
         }
         return undefined;
     }
+
+    function resolveEntityName(location: Node, node: Expression, meaning: SymbolFlags): EmitDeclarationSymbol | undefined {
+        if (isIdentifier(node)) {
+            return resolveName(location, node.escapedText, meaning);
+        }
+        else if (isPropertyAccessExpression(node) || isElementAccessExpression(node)) {
+            const symbol = resolveEntityName(location, node.expression, meaning);
+            if (symbol === undefined) return undefined;
+
+            const name = isElementAccessExpression(node) ? node.argumentExpression : node.name;
+            if (!isPropertyName(name)) return;
+
+            const memberSymbol = symbol.exports?.get(getMemberKey(name));
+            if (!memberSymbol || !(memberSymbol.flags & meaning)) {
+                return undefined;
+            }
+            return memberSymbol;
+        }
+        else {
+            return undefined;
+        }
+    }
+    function isTargetEnumDeclaration(target: Expression, location: EnumDeclaration) {

```

Comment:
Nit: Technically you're checking whether the target expression is an enum member, not a declaration.

Reply:
`isExpressionMemberOfEnum` is more descriptive I think